google-http-client/src/main/java/com/google/api/client/util/DateTime.java

```diff
@@ -269,6 +274,98 @@ public int hashCode() {
    *     time zone shift but no time.
    */
   public static DateTime parseRfc3339(String str) throws NumberFormatException {
+    return parseRfc3339WithNanoSeconds(str).toDateTime();
+  }
+
+  /**
+   * Parses an RFC3339 timestamp to a pair of seconds and nanoseconds since Unix Epoch.
+   *
+   * @param str Date/time string in RFC3339 format
+   * @throws IllegalArgumentException if {@code str} doesn't match the RFC3339 standard format; an
+   *     exception is thrown if {@code str} doesn't match {@code RFC3339_REGEX} or if it contains a
+   *     time zone shift but no time.
+   */
+  public static SecondsAndNanos parseRfc3339ToSecondsAndNanos(String str)
+      throws IllegalArgumentException {
+    return parseRfc3339WithNanoSeconds(str).toSecondsAndNanos();
+  }
+
+  /** A timestamp represented as the number of seconds and nanoseconds since Epoch. */
+  public static final class SecondsAndNanos implements Serializable {
```

Comment:
I'm not sure what our plans are for Java 8, but if we could use java.time I think we could avoid introducing this class and associated public API. @chingor13 has any final decision been made on that?

Reply:
This library is very heavily used on android, so even if we move google-cloud-java to Java 8, we'd likely avoid using new features for a long time. java.time was introduced in Android at API level 26 (Oreo).